Data load tools

anguschadney opened the pull request.

davidszotten commented on requirement.txt:
```diff
@@ -0,0 +1,3 @@
+jupyter==1.0.0
```
normally called `requirements.txt` in plural

anguschadney replied:
typo ta

davidszotten commented on analysis/load_data_example.py:
```diff
@@ -0,0 +1,90 @@
+# Add repo directory to path so that we can do the relative import below
+# Probably a better way to do this!
+import sys
+sys.path.append('/home/gus/Repos/kaggle')
```
yea, may not want to hard code paths from your local machine. to make it a bit more independent you can add folders to the python path "outside" the python process, i.e. before launching, with something like

```
$ export PYTHONPATH=<path>
$ jupyter-notebook
```

or add a minimal `setup.py` (see our other projects) and install the local project with `pip install -e`

davidszotten commented on tools/load_data.py:
```diff
@@ -0,0 +1,78 @@
+
+    def _get_path(self, data_name):
+        data_file = data_name + '.csv.zip'
+        return '/'.join([self._data_dir, data_file])
```
maybe `os.path.join`

anguschadney replied:
great thanks

davidszotten commented on tools/load_data.py:
```diff
@@ -0,0 +1,78 @@
+        self.data[data_name] = pd.read_csv(filename, **load_args)
+
+    def load_all(self, load_arg_override=None):
+        """Load all the data files into a dict of pandas dataframes.
```
the api for choosing configuration feels a bit clunky. might be worth having a think about what would be a nice way to _use_ the apis. what options would a user typically want to override? do these fall into some categories/situations of use cases?

anguschadney replied:
Yeah I see what you mean, I only ran into this problem because it took _ages_ to load the train dataset into memory, so wanted a way to limit the rows just for that dataset.  Not even that sure any of the other load options will be used...
Will have a think about a better way to do it!

davidszotten commented on tools/load_data.py:
```diff
@@ -0,0 +1,78 @@
+                                    self.data_load_args (see class docstring)
+                                    e.g. {'train': {'nrows': 2000}}
+        """
+        if load_arg_override:
```
having a method called `load` mutate the state of this object is quite confusing, and i'd call that an anti-pattern. probably worth revisiting the api (see above) before addressing this though

e.g. (currently)

```
a = gp.load_all()   # loads first 1000
b = gp.load_all(override_to_get_2k_rows)   # loads first 2000
c = gp.load_all()   # same as first call, but still loads 2000...
```

anguschadney commented:
@davidszotten I've reworked it to make the api a bit clearer, I've removed the functionality to load them all at once but I think it's a nicer this way.  What do you think?
